### ISSUE TITLE
Additional snapshot being generated

### ISSUE THREAD

**Issue by kapeels** [jimsalterjrs/sanoid#111]
This is more of a question than a bug report.

Here is my sanoid.conf:

```
#### DATASETS ####
[tank/archive]
 use_template = general

#### TEMPLATES ####
[template_general]
 hourly = 0
 daily = 30
 monthly = 18
 yearly = 2
 autosnap = yes
 autoprune = yes

###################################################################################
# default template - contains same values as hardcoded, unless you override here. #
#                    ALL values set here, so useful as documentation even if      #
#                    nothing is actually overridden.                              #
###################################################################################

[template_default]
template = yes
autoprune = yes
hourly = 48
daily = 90
monthly = 6
yearly = 0
min_percent_free = 10
autosnap = 1;
hourly_min = 0;
daily_hour = 1;
daily_min = 59;
monthly_mday = 1;
monthly_hour = 0;
monthly_min = 0;
yearly_mon = 1;
yearly_mday = 1;
yearly_hour = 0;
yearly_min = 0;
monitor = yes
monitor_dont_warn = no
monitor_dont_crit = no
hourly_warn = 90
hourly_crit = 360
daily_warn = 28
daily_crit = 32
monthly_warn = 32
monthly_crit = 35
yearly_warn = 0
yearly_crit = 0
```

And here is my partial sanoidsnapshots.txt

```
tank/archive@autosnap_2017-06-15_01:59:00_daily	creation	1497517140	-
tank/archive@autosnap_2017-06-15_02:00:00_daily	creation	1497517200	-
tank/archive@autosnap_2017-06-16_01:59:00_daily	creation	1497603540	-
tank/archive@autosnap_2017-06-16_02:00:00_daily	creation	1497603600	-
tank/archive@autosnap_2017-06-17_01:59:00_daily	creation	1497689940	-
tank/archive@autosnap_2017-06-17_02:00:00_daily	creation	1497690000	-
tank/archive@autosnap_2017-06-18_01:59:00_daily	creation	1497776340	-
tank/archive@autosnap_2017-06-18_02:00:00_daily	creation	1497776400	-
tank/archive@autosnap_2017-06-19_01:59:00_daily	creation	1497862740	-
tank/archive@autosnap_2017-06-19_02:00:00_daily	creation	1497862800	-
tank/archive@autosnap_2017-06-20_01:59:00_daily	creation	1497949140	-
tank/archive@autosnap_2017-06-20_02:00:00_daily	creation	1497949200	-
tank/archive@autosnap_2017-06-21_01:59:00_daily	creation	1498035540	-
tank/archive@autosnap_2017-06-21_02:00:00_daily	creation	1498035600	-
tank/archive@autosnap_2017-06-22_01:59:00_daily	creation	1498121940	-
tank/archive@autosnap_2017-06-22_02:00:00_daily	creation	1498122000	-
tank/archive@autosnap_2017-06-23_01:59:00_daily	creation	1498208340	-
tank/archive@autosnap_2017-06-23_02:00:00_daily	creation	1498208400	-
```

Per default template, snapshot should be created at 1:59am daily. But for some reason, an additional snapshot is generated at 2:00am daily. 
Is this behavior expected? Could somebody please explain the reason behind this?

**Comment by jimsalterjrs** [comment]
I merged the offbyone patch in #113. Can you download a fresh copy of sanoid from master and see if it fixed your issue, @kapeels?

**Comment by phreaker0** [comment]
should be fixed by #119
